js/components/chat/message-list/message-row/message-content/index.jsx

```diff
@@ -64,28 +64,32 @@ export default class MessageContent extends Component {
       .catch(console.info);
   }
 
-  render() {
-    const { message, className } = this.props;
-    const { firstUrl, isVideo, isImage, tags, hasTags } = this.state;
-
-    const hasExtraContent = !_.isEmpty(firstUrl);
+  hasExtraContent() {
+    const { firstUrl } = this.state;
+    return !_.isEmpty(firstUrl);
+  }
 
-    let extraContent = '';
+  extraContent() {
+    const { firstUrl, isVideo, isImage, tags, hasTags } = this.state;
 
-    if (hasExtraContent) {
+    if (this.hasExtraContent()) {
       if (isVideo) {
-        extraContent = <VideoContent url={firstUrl} />;
+        return <VideoContent url={firstUrl} />;
       } else if (isImage) {
-        extraContent = <ImageContent url={firstUrl} />;
+        return <ImageContent url={firstUrl} />;
       } else if (hasTags) {
-        extraContent = <UrlContent tags={tags} />;
+        return <UrlContent tags={tags} />;
       }
     }
+  }
+
+  render() {
+    const { message, className } = this.props;
 
     return (
       <span className={className}>
-        {message}
-        {extraContent}
+        {!this.hasExtraContent() && message}
```

Comment:
`this.hasExtraContent()` is called twice per render.
I'd extract it to a variable in the render method, and then remove the call from `extraContent`, and add it to the render, maybe something like: `{ hasExtraContent && this.extraContent() }`